articles/mysql-query-performance-optimization-tips/index.md

```diff
@@ -0,0 +1,180 @@
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+```sql
+SELECT ID, Name, District FROM city LIMIT 10;
+```
+
+#### MySQL Query Caching
+MySQL Query Caching provides database caching functionality. The SELECT statement text and the retrieved result are stored in the cache. When you make a similar query to the one already in the cache, MySQL will respond and give a query already in the cache. In this way, fewer resources are used, and your query runs faster.
+
+This works best with a database the more select queries are made. Once the table is updated, the cached query and result become invalid. Thus, caching may not work with an application that updates the table frequently.
+
+The command below is used to check if query cache is enabled in MySQL.
+
+```sql
+SHOW VARIABLES LIKE 'have_query_cache';
+```
+
+If the query cache is not set, set the query cache by following guidelines on [MySQL Documentation](https://dev.mysql.com/doc/refman/5.6/en/query-cache-configuration.html#:~:text=To%20set%20the%20size%20of,default%20for%20query_cache_type%20of%200.).
+
+#### Converting OUTER JOINs to INNER JOINs
+An INNER JOIN returns rows that contain columns from both tables. Unlike INNER JOIN, OUTER JOIN returns rows where no matches have been found on both tables. Therefore, OUTER JOIN does more work than inner join, increasing total execution time.
+
+Use INNER JOIN whenever possible. It would be a waste of performance to use OUTER JOIN when you don't need the data outside specified columns. We have a sample database with two tables as follows:
+1. student - student_id, first_name, last_name
+2. orders - id, date, amount, customer_id
+
+An INNER JOIN query to the table would be as it is shown below.
+
+```sql
+SELECT  C.id, C.name, O.amount, O.date FROM customers C
+  INNER JOIN orders O ON O.customer_id = C.id;
+```
+
+An OUTER JOIN query to the table would be as it is shown below.
+*NOTE: MySQL does not support FULL OUTER JOIN, but other SQL dialects such as [PostgreSQL](https://www.postgresql.org/) do.*
+
+```sql
+SELECT  C.id, C.name, O.amount, O.date FROM customers C
+  FULL OUTER JOIN orders O ON O.customer_id = C.id;
+```
+
+#### Optimize Like Statements with Union Clause
+The OR operator is used to combine two Boolean expressions and return true when either of the condition is met. When using comparison operator 'or' in a query, MySQL optimizer may incorrectly choose a full table scan to retrieve the result set. This makes the query run slower. Union clause runs faster and gives the same result.
+
+Consider the query below:
+
+```sql
+SELECT * FROM city WHERE Name like 'C%' or District LIKE 'C%';
+```
+
+Below are the optimized versions of the query above using the UNION ALL and UNION operators respectively.
+
+```sql
+SELECT * FROM city WHERE Name LIKE 'C%' UNION ALL SELECT * FROM city WHERE District LIKE 'C%';
+```
+
+```sql
+SELECT * FROM city WHERE Name LIKE 'C%' UNION SELECT * FROM city WHERE District LIKE 'C%';
+```
```

Comment:
This is a suggestion and not a requirement, but it would be nice to see a screenshot of the optimized queries running faster than the unoptimized queries. It would add legitimacy to your article.

Reply:
@mhoulmiere  I tried this, and I was not able to provide screenshots. This would require a huge/complex database to be able to show the contrasts in running time. Nevertheless, I am preparing another article on Indexes, already working on a huge database that I will use to demo differences in running time up to 4 times faster.